nucliadb/src/nucliadb/ingest/orm/processor/processor.py

```diff
@@ -0,0 +1,741 @@
+                        await self.notify_abort(
+                            partition=partition,
+                            seqid=seqid,
+                            multi=message.multiid,
+                            kbid=message.kbid,
+                            rid=message.uuid,
+                            source=message.source,
+                        )
+                        raise exc
+            finally:
+                if txn.open:
+                    if transaction_check:
+                        await sequence_manager.set_last_seqid(txn, partition, seqid)
+                    await txn.commit()
+        await self.notify_commit(
+            partition=partition,
+            seqid=seqid,
+            multi=message.multiid,
+            message=message,
+            write_type=writer_pb2.Notification.WriteType.DELETED,
+        )
+
+    @processor_observer.wrap({"type": "commit_slug"})
+    async def commit_slug(self, resource: Resource) -> None:
+        # Slug may have conflicts as its not partitioned properly,
+        # so we commit it in a different transaction to make it as short as possible
+        prev_txn = resource.txn
+        try:
+            async with self.driver.transaction() as txn:
+                resource.txn = txn
+                await resource.set_slug()
+                await txn.commit()
+        finally:
+            resource.txn = prev_txn
+
+    @processor_observer.wrap({"type": "txn"})
+    async def txn(
+        self,
+        messages: list[writer_pb2.BrokerMessage],
+        seqid: int,
+        partition: str,
+        transaction_check: bool = True,
+    ) -> None:
+        if len(messages) == 0:
+            return None
+
+        kbid = messages[0].kbid
+        if not await datamanagers.atomic.kb.exists_kb(kbid=kbid):
+            logger.info(f"KB {kbid} is deleted: skiping txn")
+            if transaction_check:
+                async with datamanagers.with_rw_transaction() as txn:
+                    await sequence_manager.set_last_seqid(txn, partition, seqid)
+                    await txn.commit()
+            return None
+
+        async with self.driver.transaction() as txn:
+            try:
+                multi = messages[0].multiid
+                kb = KnowledgeBox(txn, self.storage, kbid)
+                uuid = await self.get_resource_uuid(kb, messages[0])
+                resource: Optional[Resource] = None
+                handled_exception = None
+                created = False
+
+                for message in messages:
+                    if resource is not None:
+                        assert resource.uuid == message.uuid
+
+                    if message.source == writer_pb2.BrokerMessage.MessageSource.WRITER:
+                        resource = await kb.get(uuid)
+
+                        if resource is None:
+                            # It's a new resource
+                            resource = await kb.add_resource(uuid, message.slug, message.basic)
```

Comment:
This seems to be a new call, not sure yet why this is here :thinking: 

Edit: Ah, this was inside apply_resource before. :ok_hand: